src/Avalanche.js

```diff
@@ -146,6 +147,51 @@ export default class Avalanche {
     };
   }
 
+  /**
+   * Sign a transaction with a given BIP-32 path.
+   *
+   * @param derivation_path a path in BIP-32 format
+   * @param txn binary of the transaction
+   * @return a buffer with the signature data
+   * @example
+   * await avalanche.signTransaction(
+   *   "44'/9000'/0'/0/0",
+   *   Buffer.from("0000000000000000000000000000000000000000000000000000000000000000", "hex")
+   * );
+   */
+  async signTransaction(
+    derivation_path: string,
+    txn: Buffer,
+  ): Promise<{
+    hash: Buffer,
+    signature: Buffer
+  }> {
+    const bipPath = BIPPath.fromString(derivation_path).toPathArray();
+
+    let rawPath = Buffer.alloc(1 + bipPath.length * 4);
+    rawPath.writeInt8(bipPath.length, 0);
+    bipPath.forEach((segment, index) => {
+      rawPath.writeUInt32BE(segment, 1 + index * 4);
+    });
```

Comment:
I now have a function that builds this exact buffer which you can use along with Buffer.concat.